manifests/canonical.pp

```diff
@@ -31,12 +31,17 @@
 #   Where to create the file. If not defined "${postfix::confdir}/canonical"
 #   will be used as path.
 #
+# @param lookup_table_suffix
+#   Depends on the lookup table type, which is used on the postfix::hash and postfix::config resources.
+#   Defaults to 'db', the suffix of the "hash" type.
+#
 # @see https://www.postfix.org/canonical.5.html
 #
 define postfix::canonical (
   String                   $destination,
-  Enum['present','absent'] $ensure      = 'present',
-  Stdlib::Absolutepath     $file        = undef
+  Enum['present','absent'] $ensure              = 'present',
+  Stdlib::Absolutepath     $file                = undef,
+  String                   $lookup_table_suffix = 'db'
```

Comment:
An empty string is probably not acceptable here?  Also style: missing coma:

```suggestion
  String[1]                $lookup_table_suffix = 'db',
```

Reply:
Will add the comma, thanks for the hint.
No, empty string is not acceptable. It is used to construct the notify to the map generate exec resource, it can't be empty! So, yeah, i will add the minimum length as well, also thanks for that hint.